### ISSUE TITLE
[Bug] Betterclasses with infect type causing issues

### ISSUE THREAD

**Issue by ItzEnderArkail** [Eredrim/pvparena#147]
there's a issue where you set classes with potion effects, they become infected and return to the server spawn instead of infected spawn and get no effects

How to preduce:
set your minigame as infect type and enable betterclasses mod
set %infected% class any potion effect, kill any survivor and boolah!

Plugin Version running this issue: 1.15.2
Server Version: Paper 1.16.5

log:
```
[16:58:47 ERROR]: Could not pass event PlayerDeathEvent to pvparena v1.15.2
java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.Map.get(Object)" is null
        at net.slipcor.pvparena.modules.betterclasses.BetterClasses.parseClassChange(BetterClasses.java:513) ~[?:?]
        at net.slipcor.pvparena.loadables.ArenaModuleManager.parseClassChange(ArenaModuleManager.java:225) ~[?:?]
        at net.slipcor.pvparena.arena.ArenaPlayer.setArenaClass(ArenaPlayer.java:754) ~[?:?]
        at net.slipcor.pvparena.goals.GoalInfect.commitPlayerDeath(GoalInfect.java:483) ~[?:?]
        at net.slipcor.pvparena.classes.PACheck.handlePlayerDeath(PACheck.java:602) ~[?:?]
        at net.slipcor.pvparena.listeners.PlayerListener.onPlayerDeath(PlayerListener.java:364) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor457.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.5.jar:git-Paper-794]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callPlayerDeathEvent(CraftEventFactory.java:876) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityPlayer.die(EntityPlayer.java:818) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityLiving.damageEntity(EntityLiving.java:1362) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityHuman.damageEntity(EntityHuman.java:870) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityPlayer.damageEntity(EntityPlayer.java:982) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityHuman.attack(EntityHuman.java:1196) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityPlayer.attack(EntityPlayer.java:2022) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:2413) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.PacketPlayInUseEntity.a(PacketPlayInUseEntity.java:55) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.PacketPlayInUseEntity.a(PacketPlayInUseEntity.java:12) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:35) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1271) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1264) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:1225) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1139) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-794]
        at java.lang.Thread.run(Thread.java:831) [?:?]

```

**Comment by Eredrim** [comment]
Hello,
Can you check your arena configuration? It seems value of `modules.betterclasses.maxPlayerSwitches` is not a valid number (integer)

**Comment by Eredrim** [comment]
Fixed in 2.0
https://github.com/Eredrim/pvparena_modules/commit/8dd3a1ff20e191e6e3079bb263e3c99516af7a35
